src/modules/esl-share/core/esl-share.ts

```diff
@@ -70,17 +20,37 @@ function getButtonsList(config: ShareConfig, list: string): ShareButtonConfig[]
  */
 export class ESLShare extends ESLBaseElement {
   public static override is = 'esl-share';
-  protected static _config: Promise<ShareConfig> = Promise.reject('Configuration is not set');
+  protected static _config: Promise<ESLShareConfig>;
+  protected static _popupStore: Map<string, ESLPopup> = new Map<string, ESLPopup>();
 
   /** Register {@link ESLShare} component and dependent {@link ESLShareButton} */
   public static override register(): void {
     ESLShareButton.register();
+    ESLShareTrigger.register();
     super.register();
   }
 
+  /**
+   * Gets or updates config with a promise of a new config object or using a config provider function.
+   * @returns Promise of the current config
+   */
+  public static config(provider?: ESLShareConfigProviderType | ESLShareConfig): Promise<ESLShareConfig> {
+    if (typeof provider === 'function') ESLShare._config = provider();
+    if (typeof provider === 'object') ESLShare._config = Promise.resolve(provider);
+    return ESLShare._config ?? Promise.reject('Configuration is not set');
+  }
+
   /** Event to dispatch on ready state of {@link ESLShare} */
   @prop('esl:share:ready') public SHARE_READY_EVENT: string;
 
+  /** Default initial params to pass into the newly created popup */
+  @prop({
+    position: 'top',
+    defaultParams: {
+      hideDelay: 200
+    }
+  }) public popupInitialParams: PopupActionParams;
```

Comment:
```suggestion
  }) protected popupInitialParams: PopupActionParams;
```
Due to the follow-up reasons